### PR TITLE
Remove hard cap of 0x80 scenery items in a group

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I'm sure no one would ever exceed this but if they do well now its unlimited (it always was for json objects but would likely cause crashes)

### CODE REVIEW

File: src/openrct2/object/SceneryGroupObject.cpp [hunk 1]
```diff
@@ -118,8 +118,7 @@ void SceneryGroupObject::UpdateEntryIndexes()
         auto sceneryType = GetSceneryType(ori->Type);
         if (sceneryType.has_value())
         {
-            _legacyType.scenery_entries[_legacyType.entry_count] = { sceneryType.value(), entryIndex };
-            _legacyType.entry_count++;
+            _legacyType.SceneryEntries.push_back({ sceneryType.value(), entryIndex });
```
Comment (tupaschoal) [inline src/openrct2/object/SceneryGroupObject.cpp:121]:
```suggestion
            _legacyType.SceneryEntries.emplace_back( sceneryType.value(), entryIndex );
```

**Review by tupaschoal — APPROVED** [review]

File: src/openrct2/object/SceneryGroupObject.cpp [hunk 2]
```diff
@@ -118,8 +118,7 @@ void SceneryGroupObject::UpdateEntryIndexes()
         auto sceneryType = GetSceneryType(ori->Type);
         if (sceneryType.has_value())
         {
-            _legacyType.scenery_entries[_legacyType.entry_count] = { sceneryType.value(), entryIndex };
-            _legacyType.entry_count++;
+            _legacyType.SceneryEntries.emplace_back( sceneryType.value(), entryIndex );
```
Comment (tupaschoal) [inline src/openrct2/object/SceneryGroupObject.cpp:121]:
```suggestion
            _legacyType.SceneryEntries.emplace_back(sceneryType.value(), entryIndex);
```

File: src/openrct2/object/SceneryGroupObject.cpp [hunk 3]
```diff
@@ -118,8 +118,7 @@ void SceneryGroupObject::UpdateEntryIndexes()
         auto sceneryType = GetSceneryType(ori->Type);
         if (sceneryType.has_value())
         {
-            _legacyType.scenery_entries[_legacyType.entry_count] = { sceneryType.value(), entryIndex };
-            _legacyType.entry_count++;
+            _legacyType.SceneryEntries.emplace_back(sceneryType.value(), entryIndex);
```
Comment (tupaschoal) [inline src/openrct2/object/SceneryGroupObject.cpp:121]:
```suggestion
            _legacyType.SceneryEntries.push_back({ sceneryType.value(), entryIndex });
```

Comment (tupaschoal) [reply src/openrct2/object/SceneryGroupObject.cpp]: `ScenerySelection` is a `struct` with no user defined ctors, so emplace back did not work and thus I reverted it.

File: src/openrct2/world/Scenery.h [hunk 4]
```diff
@@ -191,8 +191,7 @@ struct rct_scenery_group_entry
 {
     rct_string_id name;
     uint32_t image;
-    ScenerySelection scenery_entries[0x80];
-    uint8_t entry_count;
+    std::vector<ScenerySelection> SceneryEntries;
     uint8_t priority;
     uint32_t entertainer_costumes;
 };
```
Comment (IntelOrca) [inline src/openrct2/world/Scenery.h:197]: This structure should be removed altogether. Everything should access the `SceneryGroupObject` directly. There probably aren't too many references to change for scenery groups.

Comment (duncanspumpkin) [reply src/openrct2/world/Scenery.h:197]: That would mean redesigning how objects work to remove the GetLegacy function which would be great to do but will take a lot of work as all the other objects need the same fixes.

**Pull request merged**